aperturedb/Collection.py

```diff
@@ -0,0 +1,149 @@
+from __future__ import annotations
+import pandas as pd
+
+
+class Collection(pd.DataFrame):
+    """
+    **Collection is a set of homogenous objects retrieved from a repository**
+
+    Collection can be further queried, ordered and split (as a DataFrame).
+    It can also be combined with another Collection using set operations.
+
+
+    .. note::
+        - Collection is essentially DataFrame with some overrides and added functionality.
+        - Most of the methods here work on the cached data, except retrieve_* and display
+    """
+    _metadata = []
+
+    def __init__(self, data: dict) -> None:
+        super().__init__(data)
+
+    @property
+    def _constructor(self):
+        return Collection
+
+    def show_stats(self) -> None:
+        """
+        **Can be used to show stats about the collection in notebook**
+        """
+        pass
+
+    def display(self,
+                show_bounding_boxes: bool = False,
```

Comment:
shouldn't this parameters be only present for "Image" collections? they make no sense for a Entity collection for instance.